### PR TITLE
Update error messages, minor fixes

### LINKED ISSUE #169: Some special names cannot be accepted
Regex is too restrictive, certain special characters, ö, is not acceptable.

### PR DESCRIPTION
**Pull request by ZShunRen**
Made the following changes:
1. Updated hint message for add command, to use `r/` prefix instead of the old `rn/`.
2. Updated user guide to reflect changes in `add` command.
3. Added a more useful hint to edit command, when a valid index is given, but no flags are specified.
4. Updated name constraints, to allow for letters, numbers, spaces, hyphens and apostrophes only, in both code and user guide.
5. Updated the hint message for `add` message, to follow the same order as laid out in the example usage.
6. Updated `add`, `edit` and `filter` command to allow for showing of multiple invalid flag values.
7. Updated user guide to reflect changes in `edit` and `filter` command, and more generally, changed messages in user guide to match those seen in product.

Closes #169, #167, #164, #165, #170, #183

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -320,14 +320,14 @@ add n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOBNAME> i/ <INCOME> [t/ <T
...
+Each client's record includes their name, contact number, email, occupation, and income. You can also enter the optional fields for credit card tier and remark(s) here. Otherwise, new users are assigned a default value of "NA".
 
 **Command Format:**
 ```
-add n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOBNAME> i/ <INCOME> [t/ <TIER>] [rn/ <REMARK>]
+add n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOBNAME> i/ <INCOME> [t/ <TIER>] [r/ <REMARK>]
 ```
 * Mandatory Fields: `n/`, `p/`, `e/`, `a/`, `j/`, `i/`
-* Optional Fields: `t/`, `rn/`
+* Optional Fields: `t/`, `r/`
```
Comment (FionaQY) [inline docs/UserGuide.md:330]: I think `s/` should also be added here.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -336,9 +336,9 @@ For detailed explanations of each flag and acceptable arguments, refer to Sectio
   ```
   add n/ JOHN DOE p/ 99007766 e/ mrdoe@ntu.sg a/ com3 j/ doctor i/ 99999
   ```
-- Add new client with tier and remark:
+- Add new client with tier and remark(s):
   ```
-  add n/ JOHN DOE p/ 99007766 e/ mrdoe@ntu.sg a/ com3 j/ doctor i/ 99999 t/ gold rn/ got anger issue
+  add n/ JOHN DOE p/ 99007766 e/ mrdoe@ntu.sg a/ com3 j/ doctor i/ 99999 t/ gold r/ got anger issue
```
Comment (FionaQY) [inline docs/UserGuide.md:341]: just for clarity's sake, should we add a "s/ urgent" here to show all the prefixes users can use?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -347,12 +347,12 @@ For detailed explanations of each flag and acceptable arguments, refer to Sectio
...
       New client added: Name: <NAME>, Phone: <PHONE>, Email: <EMAIL>, Address: <ADDRESS>, Job: <JOB>, Income: <INCOME>, Tier: <TIER>, Remark: <REMARK>.
       ```
-    - If "Tier" and "Remark" are not provided, they will be set to "N.A." and displayed as such in the success message.
+    - If "Tier" and "Remark" are not provided, they will be set to "NA" and displayed as such in the success message.
 
 - **On Error**
     - Message:
       ```
-      Please verify that your input is in the correct format. Include the following details: n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOBNAME> i/ <INCOME> [t/ <TIER>] [rn/ <REMARK>].
+      Please verify that your input is in the correct format. Include the following details: n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOBNAME> i/ <INCOME> [t/ <TIER>] [r/ <REMARK>].
```
Comment (FionaQY) [inline docs/UserGuide.md:355]: I believe [s/ Status] should be included as well.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -376,15 +376,15 @@ For detailed explanations of each flag and acceptable arguments, refer to Sectio
...
 
-All client information, including contact details, address, job information, and other relevant data, can be modified. You can also append to or replace existing remarks and adjust the client's tier status.
+All client information, including contact details, address, job information, and other relevant data, can be modified. You can also append to or replace existing remark(s) and adjust the client's tier status.
 
 **Command Format:**
 ```
 edit <INDEX> n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOB> i/ <INCOME> [t/ <TIER>] [rn/ <NEW REMARK>] [ra/ <REMARK TO BE APPENDED>]
 ```
 - Mandatory Field: `<INDEX>`
 - Optional Fields: `n/`, `p/`, `e/`, `a/`, `j/`, `i/`, `t/`, `rn/`, `ra/`
```
Comment (FionaQY) [inline docs/UserGuide.md:386]: ditto

File: docs/UserGuide.md [hunk 5]
```diff
@@ -532,7 +532,7 @@ list
 
 **Command Format:**
 ```
-filter n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOB> r/ <REMARKS> t/ <TIER> i/ <INCOME>
+filter n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOB> r/ <REMARK> t/ <TIER> i/ <INCOME>
```
Comment (FionaQY) [inline docs/UserGuide.md:535]: ditto

File: docs/UserGuide.md [hunk 6]
```diff
@@ -596,7 +596,7 @@ For detailed explanations of each flag and acceptable arguments, refer to Sectio
   
       Parameters: <FLAG>/ <SEARCH TERM>
     
-      Flags: n/ (name), p/ (phone), e/ (email), a/ (address), j/ (job), r/ (remarks)
+      Flags: n/ (name), p/ (phone), e/ (email), a/ (address), j/ (job), r/ (remark)
```
Comment (FionaQY) [inline docs/UserGuide.md:599]: ditto

**Review by FionaQY — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 7]
```diff
@@ -727,6 +727,7 @@ Each credit card tier is visually distinguished in the UI: Gold is marked with a
 
 1. **When using multiple screens**, if you move the application to a secondary screen, and later switch to using only the primary screen, the GUI will open off-screen. The remedy is to delete the `preferences.json` file created by the application before running the application again.
 2. **If you minimize the Help Window** and then run the `help` command (or use the `Help` menu, or the keyboard shortcut `F1`) again, the original Help Window will remain minimized, and no new Help Window will appear. The remedy is to manually restore the minimized Help Window.
+3. **When inputting names that use `/`**, this can cause an `invalid command` error to be shown or even unintended attributes for the client added. Avoid using `/`, and spell out the names in full, for cases like 'Ramesh s/o Ravichandran', change it to 'Ramesh Son Of Ravichandran'
```
Comment (iamdiluxedbutcooler) [inline docs/UserGuide.md:730]: I still having concern if banking agent should record the name as per officials', where s/o is allowed in official/gov document. So im thinking should we at least show on the list name with s/o (but then when user gg to search with cli it still needs to search by Son Of not s/o). Might have better way to mitigate/or defend our point about that but We can discuss about this more later

Comment (iamdiluxedbutcooler) [reply docs/UserGuide.md:730]: I came up with some roughly draft pseudocode on the checking logic for s/o and d/o appeared under n/ flag in add and edit command, can consider this change if we have enough time in this iteration

```
    static final List<String> VALID_COMMANDS = ......;
    static final List<String> NAME_SUFFIXES = Arrays.asList("s/o", "d/o");
    
    public List<Command> parse(String input) {
        List<Command> commands = new ArrayList<>();
        int currentPos = 0;
        
        while (currentPos < input.length()) {
            String nextCommand = findNextCommand(input, currentPos);
            if (nextCommand == null) break;
            
            int commandStart = input.indexOf(nextCommand, currentPos);
            
            if (nextCommand.equals("n/")) {
                int nextRealCommand = findNextRealCommand(input, commandStart + nextCommand.length());
                String content = nextRealCommand != -1 
                    ? input.substring(commandStart + nextCommand.length(), nextRealCommand).trim()
                    : input.substring(commandStart + nextCommand.length()).trim();
                commands.add(new NameCommand(content));
                currentPos = nextRealCommand != -1 ? nextRealCommand : input.length();
            } else {
                int contentEnd = findContentEnd(input, commandStart + nextCommand.length());
                String content = input.substring(commandStart + nextCommand.length(), contentEnd).trim();
                commands.add(createCommand(nextCommand, content));
                currentPos = contentEnd;
            }
        }
        return commands;
    }
    
    private int findNextRealCommand(String input, int startPos) {
        int pos = startPos;
        while (pos < input.length()) {
            // find next slash
            pos = input.indexOf("/", pos);
            if (pos == -1) return -1;
            
            // check if it's part of the special name suffix
            boolean isNameSuffix = false;
            for (String suffix : NAME_SUFFIXES) {
                // check the slash if it's for s/o or d/o or not
                if (pos >= 1 && pos + 1 < input.length() && 
                    suffix.equals(input.substring(pos - 1, pos + 2))) {
                    isNameSuffix = true;
                    break;
                }
            }
            
            if (isNameSuffix) {
                pos += 2;
                continue;
            }
            
            // found a real command prefix
            if (pos >= 1 && VALID_COMMANDS.contains(input.substring(pos - 1, pos + 1))) {
                return pos - 1;
            }
            pos++;
        }
        return -1;
    }

```

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 8]
```diff
@@ -49,8 +49,8 @@ public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_JOB,
                         PREFIX_INCOME, PREFIX_TIER, PREFIX_REMARK, PREFIX_STATUS);
-        PrefixCheckResult prefixCheckResult = arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS,
-                PREFIX_JOB, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_INCOME);
+        PrefixCheckResult prefixCheckResult = arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
```
Comment (iamdiluxedbutcooler) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:52]: good catch on the ordering!

File: src/main/java/seedu/address/model/client/Name.java [hunk 9]
```diff
@@ -10,13 +10,13 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain letters, numbers, spaces, hyphens, apostrophes, and it should not be blank";
```
Comment (iamdiluxedbutcooler) [inline src/main/java/seedu/address/model/client/Name.java:13]: wanna add "no symbols" too to match explaination on the UG?

**Review by iamdiluxedbutcooler — APPROVED** [review]
LGTM! Minor concern can be discussed later on!

File: docs/UserGuide.md (hunk 1 above)
Comment (iamdiluxedbutcooler) [reply docs/UserGuide.md:330]: good catch!

File: docs/UserGuide.md (hunk 6 above)
Comment (iamdiluxedbutcooler) [reply docs/UserGuide.md:599]: https://www.youtube.com/watch?v=pSUydWEqKwE @FionaQY

File: docs/UserGuide.md (hunk 2 above)
Comment (ZShunRen) [reply docs/UserGuide.md:341]: Good idea! I'll implement it

File: docs/UserGuide.md (hunk 1 above)
Comment (ZShunRen) [reply docs/UserGuide.md:330]: good catch, my bad for forgetting about the feature I added myself...

File: src/main/java/seedu/address/model/client/Name.java (hunk 9 above)
Comment (ZShunRen) [reply src/main/java/seedu/address/model/client/Name.java]: Nice idea, i have added it!

**Pull request merged**